apis/kueue/v1beta1/resourceflavor_types.go

```diff
@@ -72,7 +72,7 @@ type ResourceFlavorSpec struct {
 	// +optional
 	// +listType=atomic
 	// +kubebuilder:validation:MaxItems=8
-	// +kubebuilder:validation:XValidation:rule="self.all(x, x.effect in ['NoSchedule', 'PreferNoSchedule', 'NoExecute'])", message="supported taint effect values: 'NoSchedule', 'PreferNoSchedule', 'NoExecute'"
+	// +kubebuilder:validation:XValidation:rule="self.all(x, x.effect in ['NoSchedule', 'NoExecute'])", message="supported taint effect values: 'NoSchedule' and 'NoExecute'"
```

Comment:
Indeed, setting `PreferNoSchedule` is no-op in Kueue, but I imagine people may have some third party controllers which read the nodes (as TAS does) and transform them into the resource flavors. Then, such a controller would need to have a custom logic not to set this value. 

I'm not sure people write such controllers already, but maybe it is enough to document that PreferNoSchedule is no-op from Kueue perspective? WDYT?

Reply:
Uhm, good point.
Additionally, I could imagine the situation where they already accidentally deployed RF with `PreferNoSchedule` effect. In that case, the removal of such RF will fail since kueue will fail to drop `in-use` finalizer.

Let me convert this PR to just add API comments.

Reply:
/hold